synapse/models/risk.py

```diff
@@ -0,0 +1,125 @@
+                    'doc': 'An instance of an actor attacking a target.'
+                }),
+            ),
+            'forms': (
+                ('risk:vuln', {}, (
+                    ('name', ('str', {}), {
+                        'doc': 'A user specified name for the vulnerability',
+                    }),
+
+                    ('type', ('str', {}), {
+                        'doc': 'A user specified type for the vulnerability',
+                    }),
+                    ('desc', ('str', {}), {
+                        'doc': 'A description of the vulnerability',
+                    }),
+                    ('cve', ('it:sec:cve', {}), {
+                        'doc': 'The CVE ID of the vulnerability.',
+                    }),
+                )),
+
+                ('risk:hasvuln', {}, (
+                    ('vuln', ('risk:vuln', {}), {
+                        'doc': 'The vulnerabilty present in the target.'
+                    }),
+                    ('person', ('ps:person', {}), {
+                        'doc': 'The vulnerable person',
+                    }),
+                    ('org', ('ou:org', {}), {
+                        'doc': 'The vulnerable org',
+                    }),
+                    ('host', ('it:host', {}), {
+                        'doc': 'The vulnerable host',
+                    }),
+                    ('place', ('geo:place', {}), {
+                        'doc': 'The vulnerable place',
+                    }),
+                    ('software', ('it:prod:softver', {}), {
+                        'doc': 'The vulnerable software',
+                    }),
+                    ('spec', ('mat:spec', {}), {
+                        'doc': 'The vulnerable material specification',
+                    }),
+                    ('item', ('mat:item', {}), {
+                        'doc': 'The vulnerable material item',
+                    }),
+                )),
+
+                ('risk:attacked', {}, (
+                    ('time', ('time', {}), {
+                        'doc': 'Set if the time of the attack is known',
+                    }),
+                    ('success', ('bool', {}), {
+                        'doc': 'Set if the attack was known to have succeed or not',
+                    }),
+                    ('targeted', ('bool', {}), {
+                        'doc': 'Set if the attack was assessed to be targeted or not',
+                    }),
+                    ('campaign', ('ou:campaign', {}), {
+                        'doc': 'Set if the attack was part of a larger campaign',
+                    }),
+                    ('actor:org', ('ou:org', {}), {
+                        'doc': 'The org that carried out the attack',
+                    }),
+                    ('actor:person', ('ps:person', {}), {
+                        'doc': 'The person that carried out the attack',
+                    }),
+                    ('target:org', ('ou:org', {}), {
```

Comment:
taget -> ndef ?
target:form breakout ?

Reply:
there is a disjoint between "things that are vulnerable" vs "things that are targeted"

Reply:
there has to be.  software is vulnerable but a host is targeted.